core/cluster.go

```diff
@@ -0,0 +1,114 @@
+package core
+
+import (
+	"context"
+	"time"
+
+	util "github.com/ipfs/go-ipfs-util"
+	ipfscluster "github.com/ipfs/ipfs-cluster"
+	capi "github.com/ipfs/ipfs-cluster/api"
+	"github.com/ipfs/ipfs-cluster/consensus/raft"
+	"github.com/ipfs/ipfs-cluster/monitor/pubsubmon"
+	"github.com/ipfs/ipfs-cluster/observations"
+	peer "github.com/libp2p/go-libp2p-peer"
+	"github.com/textileio/go-textile/cluster"
+)
+
+func (t *Textile) clusterExists() bool {
+	return util.FileExists(cluster.ConfigPath(t.repoPath))
+}
+
+// startCluster creates all the necessary things to produce the cluster object
+func (t *Textile) startCluster() error {
+	cfgMgr, cfgs, err := cluster.MakeAndLoadConfigs(t.repoPath)
+	if err != nil {
+		return err
+	}
+	defer cfgMgr.Shutdown()
+
+	cfgs.ClusterCfg.LeaveOnShutdown = false
+
+	tracker, err := cluster.SetupPinTracker(
+		"map",
+		t.node.PeerHost,
+		cfgs.MaptrackerCfg,
+		cfgs.StatelessTrackerCfg,
+		cfgs.ClusterCfg.Peername,
+	)
+	if err != nil {
+		return err
+	}
+
+	informer, alloc, err := cluster.SetupAllocation(
+		"disk-freespace",
+		cfgs.DiskInfCfg,
+		cfgs.NumpinInfCfg,
+	)
+	if err != nil {
+		return err
+	}
+
+	ipfscluster.ReadyTimeout = raft.DefaultWaitForLeaderTimeout + 5*time.Second
```

Comment:
I think you can ignore this (like set to 5 seconds or leave default). Raft is not involved anyways.

Reply:
👌